lib/reader/src/parser.rs

```diff
@@ -40,11 +40,32 @@ pub fn parse_test<'a>(text: &'a str) -> Result<TestFile<'a>> {
     let mut parser = Parser::new(text);
     // Gather the preamble comments as 'Function'.
     parser.gather_comments(AnyEntity::Function);
+
+    let commands = parser.parse_test_commands();
+    let mut isa_spec = parser.parse_isa_specs()?;
+
+    if let isaspec::IsaSpec::Some(ref mut isa_vec) = isa_spec {
+        if isa_vec.len() == 1 {
+            // Remove it from the isa_vec just while we parse functions, avoiding a .clone()
+            parser.target_isa = Some(isa_vec.remove(0));
```

Comment:
Ugh, yeah you would need a whole other lifetime on `Parser`.

Another option would be to pass `unique_isa` as an argument to `parse_function_list()` and `parse_function()`. Then you can stick it in the `Context`.